src/event-replay/parquet-based/gen-ids-file.ts

```diff
@@ -0,0 +1,44 @@
+import * as fs from 'fs';
+
+import { logger } from '../../logger';
+import { DatasetStore } from './dataset/store';
+import { splitIntoChunks } from './helpers';
+
+(async () => {
+  const args = process.argv.slice(2);
+  const workers: number = Number(args[0].split('=')[1]);
+
+  logger.info({ component: 'event-replay' }, `Generating ID files for ${workers} parallel workers`);
+
+  const dir = './events/new_block';
+
+  const dataset = DatasetStore.connect();
+
+  const ids: number[] = await dataset.newBlockEventsIds();
+  const batchSize = Math.ceil(ids.length / workers);
+  const chunks = splitIntoChunks(ids, batchSize);
+
+  const files = fs.readdirSync(dir).filter(f => f.endsWith('txt'));
+
+  // delete previous files
+  files.map(
+    file =>
+      new Promise((resolve, reject) => {
```

Comment:
What's the reason for wrapping this calls in a promise? They aren't awaited so they are dangling/unhandled promises.